src/test/java/teammates/test/cases/webapi/PublishFeedbackSessionActionTest.java

```diff
@@ -23,13 +30,90 @@ protected String getRequestMethod() {
     @Test
     @Override
     protected void testExecute() throws Exception {
-        // TODO
+        ______TS("Typical case");
+
+        CourseAttributes course = typicalBundle.courses.get("typicalCourse1");
+        FeedbackSessionAttributes session = typicalBundle.feedbackSessions.get("session1InCourse1");
+
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, course.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session.getFeedbackSessionName(),
+        };
+
+        PublishFeedbackSessionAction publishFeedbackSessionAction = getAction(params);
+
+        JsonResult result = getJsonResult(publishFeedbackSessionAction);
+        FeedbackSessionData feedbackSessionData = (FeedbackSessionData) result.getOutput();
+
+        assertEquals(feedbackSessionData.getFeedbackSessionName(), session.getFeedbackSessionName());
+        assertEquals(feedbackSessionData.getPublishStatus(), FeedbackSessionPublishStatus.PUBLISHED);
+
+        ______TS("Failure case: Session is already published");
+        // Attempt to publish the same session again.
+        assertTrue(logic.getFeedbackSession(session.getFeedbackSessionName(), course.getId()).isPublished());
+
+        result = getJsonResult(getAction(params));
+        MessageOutput output = (MessageOutput) result.getOutput();
+
+        assertEquals(output.getMessage(), "Error publishing feedback session: Session has already been published.");
+        assertEquals(result.getStatusCode(), HttpStatus.SC_BAD_REQUEST);
+    }
+
+    @Test
+    public void testExecute_invalidRequests_shouldFail() {
+        ______TS("non existent session name");
+
+        CourseAttributes course = typicalBundle.courses.get("typicalCourse1");
+        FeedbackSessionAttributes session = typicalBundle.feedbackSessions.get("session1InCourse1");
+        String randomSessionName = "randomName";
+        assertNotNull(course.getId(), session.getFeedbackSessionName());
```

Comment:
What does the assertion try to do here?

BTW, the first parameter of assertNotNull is `message` not `object`. There is no such assertion method for asserting two non-null objects.